compare/compare.c

```diff
@@ -593,14 +609,10 @@ int main(int argc, char *argv[])
     ItemList list = {0};
     int result = R_ERROR;
 
-    int scale = 1;
-    if (pos[SCALE]) {
-        const char *arg = argv[pos[SCALE]];
-        if (arg[0] < '1' || arg[0] > '8' || arg[1]) {
-            printf("Invalid scale value, should be 1-8!\n");
-            goto end;
-        }
-        scale = arg[0] - '0';
+    int scale_x = 1, scale_y = 1;
+    if (pos[SCALE] && !parse_scale(argv[pos[SCALE]], &scale_x, &scale_y)) {
+        printf("Invalid scale value, should be 1-8!\n");
```

Comment:
This message isn't completely correct too, should be something like `1-8[x1-8]`.

Reply:
Suggestion applied.